man/systemd-nspawn.xml

```diff
@@ -399,6 +399,18 @@
         is read from it and automatically used, also as formatted hexadecimal characters.</para></listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><option>--root-hash-sig=</option></term>
+
+        <listitem><para>Takes a path to a PKCS7 formatted signature of the <option>--root-hash=</option> option.
+        If libcryptsetup (minimum version 2.3.0) and the kernel (minimum version 5.4 with CONFIG_DM_VERITY_VERIFY_ROOTHASH_SIG
```

Comment:
We generally document required versions in README, not in the docs, as this is primarily relevant for OS vendors, and they should make sure that things "just work", and thus this is just noise...

Reply:
I'll update the README as the first thing, thanks.

I put it in the manpage as it seemed relevant for users as well. This was my reasoning: given it has to be a hard error to supply a signature when it's not supported by the combination of kernel + userspace, some users are bound to find out the hard way at some point. By stating the requirements clearly in the doc, we can give a clear heads-up and can save some headaches.

Is there an alternative way of being more user-friendly? Maybe setting the option should be rejected outright if the requirements are not met at runtime?

Reply:
We assume that version incompatibilities is something distros deal with, and we see that as noise in our user-facing docs. If distros decide to combine old stuff with new stuff, it's on *them* to document that and say what doesn't work and is a mess. We assume that systemd, kernels, and so on are either updated in lockstep (which then requires no such documentation) or people are smart enough to know what'll work and what will not.

So, please place this in README (i.e. maintainer-focussed docs) rather than the man page (i.e. user-focussed docs).

Reply:
Ok, will do.